src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/CoursesControllerTest.kt

```diff
@@ -40,199 +41,229 @@ class CoursesControllerTest(
   @MockkBean
   private lateinit var coursesService: CourseService
 
-  @Test
-  fun `get all courses`() {
-    every { coursesService.allCourses() } returns repository.allCourses()
-
-    mockMvc.get("/courses") {
-      accept = MediaType.APPLICATION_JSON
-      header(AUTHORIZATION, jwtAuthHelper.bearerToken())
-    }.andExpect {
-      status { isOk() }
-      content {
-        this.contentType(MediaType.APPLICATION_JSON)
-        this.json(
-          """
+  @Nested
```

Comment:
Would you mind adding a line to the commit message that this commit also restructures the `coursesControllerTest` to group tests by function?

Reply:
Done